gc/base/Bits.hpp

```diff
@@ -43,6 +44,34 @@ class MM_Bits {
 		return x << OMR_LOG_POINTER_SIZE;
 	}
 
+	/**
+	 * log2(size of an object to object reference)
+	 *
+	 * @param[in] compressed true if object to object references are compressed, false if not
+	 * @return the shift value
+	 */
+	MMINLINE static uintptr_t logReferenceSize(bool compressed) { return compressed ? 2 : OMR_LOG_POINTER_SIZE; }
+
+	/**
+	 * Calculate the difference between two object slot pointers, in slots
+	 *
+	 * @param[in] p1 the value to be subtracted from
+	 * @param[in] p2 the value to be subtracted
+	 * @param[in] compressed true if object to object references are compressed, false if not
+	 * @return p1 - p2 in slots
+	 */
+	MMINLINE static intptr_t subtractSlotPointers(fomrobject_t *p1, fomrobject_t *p2, bool compressed) { return ((intptr_t)p1 - (intptr_t)p2) >> logReferenceSize(compressed); }
```

Comment:
I am not sure math for signed would work for addresses potentially use high bit

Reply:
Changed to uint and added a subtract from helper to handle the negative case.